aptos-move/framework/aptos-framework/sources/object.move

```diff
@@ -0,0 +1,433 @@
+        let object = borrow_global_mut<Object>(ref.self.inner);
+        event::emit_event(
+            &mut object.transfer_events,
+            TransferEvent {
+                object_id: ref.self,
+                from: object.owner,
+                to,
+            },
+        );
+        object.owner = to;
+    }
+
+    public entry fun entry_transfer(
+        owner: &signer,
+        object_id: address,
+        to: address,
+    ) acquires Object {
+        transfer(owner, ObjectId { inner: object_id }, to)
+    }
+
+    public fun transfer(
+        owner: &signer,
+        object_id: ObjectId,
+        to: address,
+    ) acquires Object {
+        let owner_address = signer::address_of(owner);
+        verify_ungated_and_descendant(owner_address, object_id.inner);
+
+        let object = borrow_global_mut<Object>(object_id.inner);
+        if (object.owner == to) {
+            return
+        };
+
+        event::emit_event(
+            &mut object.transfer_events,
+            TransferEvent {
+                object_id: object_id,
+                from: object.owner,
+                to,
+            },
+        );
+        object.owner = to;
+    }
+
+    public fun transfer_to_object(
+        owner: &signer,
+        object_id: ObjectId,
+        to: ObjectId,
+    ) acquires Object {
+        transfer(owner, object_id, to.inner)
+    }
+
+    /// This checks that the destination address is eventually owned by the owner and that each
+    /// object between the two allows for ungated transfers.
+    fun verify_ungated_and_descendant(owner: address, destination: address) acquires Object {
+        let current_address = destination;
+        assert!(
+            exists<Object>(current_address),
+            error::not_found(EOBJECT_DOES_NOT_EXIST),
+        );
+
+        while (owner != current_address) {
+            // At this point, the first object exists and so the more likely case is that the
+            // object's owner is not an object. So we return a more sensible error.
+            assert!(
+                exists<Object>(current_address),
+                error::permission_denied(ENOT_OBJECT_OWNER),
+            );
+            let object = borrow_global<Object>(current_address);
+            assert!(
+                object.allow_ungated_transfer,
+                error::permission_denied(ENO_UNGATED_TRANSFERS),
+            );
+
+            current_address = object.owner;
+        };
+    }
+
+    /// Accessors
+
+    public fun owner(object_id: ObjectId): address acquires Object {
+        assert!(
+            exists<Object>(object_id.inner),
+            error::not_found(EOBJECT_DOES_NOT_EXIST),
+        );
+        borrow_global<Object>(object_id.inner).owner
+    }
+
+    public fun is_owner(object_id: ObjectId, owner: address): bool acquires Object {
+        owner(object_id) == owner
+    }
+
+    native fun create_signer(addr: address): signer;
```

Comment:
This will be replaced by the standalone create_signer module right?

Reply:
yes